#  <p style="text-align: center;">Практическое задание к Семинару 3.<br><br>Урок 3. EDA (exploratory data analysis) или Разведочный анализ.</p>

In [10]:
from math import factorial, exp

In [11]:
import numpy as np

**Формула вероятности:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>
$\displaystyle P(A)=\frac{m}{n}$ (1)<br>, где $m$-число благоприятных исходов, $n$-общее число всех исходов.</b></font></div>

**Формула подсчета количества сочетаний по $k$ элементов из множества $n$:**  
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$ (2)</b></font></div>

**Формула Бернулли:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle P_{n}(X=k)=C^k_{n}p^{k}q^{n-k}$ (3)</b></font></div>

**Формула Пуассона:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle P_m=\frac{\lambda^m}{m!}\cdot e^{-\lambda}$ (4)<br>, где $\lambda$ - среднее количество наступления события за определенную единицу измерений, и $\lambda=p\cdot n$.</b></font></div>

Формулы:
Среднеаифметическое
СКО
Дисперсия

In [2]:
def sochetaniy(n, k): #функция расчета количества сочетаний
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
def bernylli(n, k, p): # Формула Бернулли
    q = 1 - p;
    return float(sochetaniy(n, k)*p**k*q**(n-k))

In [4]:
def pyassona (m,p,n): # Формула Пуассона
    lambd=p*n
    return (lambd**m)/factorial(m)*exp(-lambd)

## ЗАДАНИЕ 1. 
**Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

<u>***Среднее арифметическое***</u>

In [71]:
ZP = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]
total = 0      # или sum(ZP)
for i in ZP:
    total += i
mean_ar = total/len(ZP)
mean_ar                 # Средние арифметическое

65.3

<u>***Дисперсия***</u>

In [63]:
total_var = 0
for i in ZP:
    total_var += (i-mean_ar)**2
Variance_Bia = total_var/len(ZP) # Несмещенная дисперсия
Variance

950.11

In [73]:
Variance_Unb = total_var/(len(ZP)-1) # Cмещенная дисперсия
Variance_Unb

1000.1157894736842

<u>***Среднее квадратичное отклонение***</u>

In [75]:
CKO = np.sqrt(Variance_Bia) # Variance_Bia**0.5
CKO

30.823854398825596

<u>***ПРОВЕРКА С ПОМОЩЬЮ БИБЛИОТЕКИ NUMPY***</u>

In [43]:
ZP_np = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
ZP_np.shape

(20,)

<u>***Среднее арифметическое***</u>

In [44]:
ZP_np.mean() #Среднее арифметическое

65.3

<u>***Дисперсия***</u>

In [49]:
np.var(ZP_np) # Несмещенная дисперсия

950.11

In [80]:
round(np.var(ZP_np, ddof=1),3) # Cмещенная дисперсия

1000.116

<u>***Среднее квадратичное отклонение***</u>

In [81]:
round(np.std(ZP_np),3)

30.824

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Среднее арифметическое: 65.3<br>Cреднее квадратичное отклонение: 30.824<br>Несмещенная дисперсия: 950.11<br>Cмещенная дисперсия: 1000.116</font></div>

---

## ЗАДАНИЕ 2. 
**В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

Решаем по формуле 1, $\displaystyle P(A)=\frac{m}{n}$, т.к. из условия задачи имеются разные ящики, то вероятность будем рассчитывать для каждого ящика по отдельности. 
Общая вероятность будет находиться по формуле: $P(1,2,3)=P(1) + P(2) + P(3)$, для этого:<br>
**1 сценарий (P(1))** - Из первого ящика вынули 2 белых мяча, тогда из второго 1 белый мяч:<br>
1.1) Находим все благоприятные исходы в первом ящике (все сочетания белых мячей).  => $\displaystyle C^2_{5}$<br>
1.2) Находим все возможные исходы в первом ящике (все сочетания мячей).  => $\displaystyle C^2_{8}$<br>
1.3) Находим вероятность вытащить оба мяча подряд из первого ящика. => $P(A) = \frac {C^2_{5}} {C^2_{8}}$<br>
1.4) Находим все благоприятные исходы во втором ящике (вытащить 1 белый мяч).  => $\displaystyle C^1_{5}$<br>
1.5) Находим все возможные исходы во втором ящике (все сочетания мячей).  => $\displaystyle C^4_{12}$<br>
1.6) Находим вероятность вытащить оба не белых мяча подряд из второго ящика. => $P(B) = \frac {C^2_{2}} {C^2_{11}}$<br>
1.7) Находим вероятность того, что все мячи белые. => $P(1_{(A+B)})=P(A)\cdot P(B)$

**2 сценарий (P(2))** - Из каждого ящика вынули только 1 белый мяч и 1 НЕ белый мяч:<br>
2.1) Находим все благоприятные исходы в первом ящике (все сочетания белых мячей).  => $\displaystyle C^1_{7}$<br>
2.2) Находим все благоприятные исходы в первом ящике (все сочетания НЕ белых мячей).  => $\displaystyle C^1_{3}$<br>
2.3) Находим все возможные исходы в первом ящике (все сочетания мячей).  => $\displaystyle C^2_{10}$<br>
2.4) Находим вероятность вытащить один белый и один НЕ белый мяч из первого ящика. => $P(A) = \frac {C^1_{7}+C^1_{3}} {C^2_{10}}$<br>
2.5) Находим все благоприятные исходы во втором ящике (все сочетания белых мячей).  => $\displaystyle C^1_{9}$<br>
2.6) Находим все благоприятные исходы во втором ящике (все сочетания НЕ белых мячей).  => $\displaystyle C^1_{2}$<br>
2.7) Находим все возможные исходы во втором ящике (все сочетания мячей).  => $\displaystyle C^2_{11}$<br>
2.8) Находим вероятность вытащить один белый и один НЕ белый мяч из второго ящика. => $P(B) = \frac {C^1_{9}+C^1_{2}} {C^2_{11}}$<br>
2.9) Находим вероятность того, что все мячи белые. => $P(2_{(A+B)})=P(A)\cdot P(B)$

**3 сценарий (P(3))** - Из второго ящика вынули 2 белых мяча, тогда из первого 0 белых мячей:<br>
3.1) Находим все благоприятные исходы в первом ящике (все сочетания белых мячей).  => $\displaystyle C^2_{3}$<br>
3.2) Находим все возможные исходы в первом ящике (все сочетания мячей).  => $\displaystyle C^2_{10}$<br>
3.3) Находим вероятность вытащить оба мяча подряд из первого ящика. => $P(A) = \frac {C^2_{3}} {C^2_{10}}$<br>
3.4) Находим все благоприятные исходы во втором ящике (все сочетания белых мячей).  => $\displaystyle C^2_{9}$<br>
3.5) Находим все возможные исходы во втором ящике (все сочетания мячей).  => $\displaystyle C^2_{11}$<br>
3.6) Находим вероятность вытащить оба не белых мяча подряд из второго ящика. => $P(B) = \frac {C^2_{9}} {C^2_{11}}$<br>
3.7) Находим вероятность того, что все мячи белые. => $P(3_{(A+B)})=P(A)\cdot P(B)$<br><br>

Находим вероятность того, что ровно два белых мяча:<br> 1 сценарий **ИЛИ** 2 сценарий **ИЛИ** 3 сценарий => **ИЛИ** = "+"<br>
P(1,2,3) = 1 сценарий **+** 2 сценарий **+** 3 сценарий
